std/string.d

```diff
@@ -2181,6 +2181,13 @@ auto representation(Char)(Char[] s) @safe pure nothrow @nogc
  * See_Also:
  *      $(XREF uni, toCapitalized) for a lazy range version that doesn't allocate memory
  */
+auto capitalize(S)(auto ref S s)
+    if (!isSomeString!S)
```

Comment:
This constraint will match almost everything and delay error messages into the internals of phobos. I think you should add the `is(StringTypeOf!S)` condition.